integration-tests/gradle/src/test/java/io/quarkus/gradle/KotlinIsIncludedInQuarkusJarTest.java

```diff
@@ -0,0 +1,47 @@
+package io.quarkus.gradle;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Enumeration;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+import org.junit.jupiter.api.Test;
+
+public class KotlinIsIncludedInQuarkusJarTest extends QuarkusGradleWrapperTestBase {
```

Comment:
in my fork, I created another branch to validate the purpose of this test by removing the included changes in `ProjectDescriptorBuilder`, the result is:
```
2024-09-17T02:32:37.0846761Z org.opentest4j.AssertionFailedError: Expected entry basic-kotlin-application-project/src/main/kotlin/org/acme/MyMainClass.class not found in JAR file. ==> expected: <true> but was: <false>
2024-09-17T02:32:37.0849523Z 	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
2024-09-17T02:32:37.0851336Z 	at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
2024-09-17T02:32:37.0853003Z 	at org.junit.jupiter.api.AssertTrue.failNotTrue(AssertTrue.java:63)
2024-09-17T02:32:37.0854351Z 	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:36)
2024-09-17T02:32:37.0855760Z 	at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:214)
2024-09-17T02:32:37.0857660Z 	at io.quarkus.gradle.KotlinIsIncludedInQuarkusJarTest.assertJarContainsEntry(KotlinIsIncludedInQuarkusJarTest.java:45)
2024-09-17T02:32:37.0860012Z 	at io.quarkus.gradle.KotlinIsIncludedInQuarkusJarTest.testFastJarFormatWorks(KotlinIsIncludedInQuarkusJarTest.java:28)
2024-09-17T02:32:37.0861883Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
2024-09-17T02:32:37.0863187Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2024-09-17T02:32:37.0864658Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2024-09-17T02:32:37.0866574Z 
```

[Log file](https://productionresultssa10.blob.core.windows.net/actions-results/7730bb96-5f1e-470a-b67a-d5eacc46d0d2/workflow-job-run-d902fd06-6a09-54f6-29ea-e4a86af45f77/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-09-17T03%3A15%3A43Z&sig=yXMHgSg5sOWGi2dYA%2FpbegOy43%2FJOpCiRLqP%2FVDLHic%3D&ske=2024-09-17T14%3A41%3A46Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-09-17T02%3A41%3A46Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-05-04&sp=r&spr=https&sr=b&st=2024-09-17T03%3A05%3A38Z&sv=2024-05-04)